tools/build_defs/old_12_2018/framework.bzl

```diff
@@ -266,21 +266,46 @@ def cc_external_rule_impl(ctx, attrs):
     ]
 
     script_text = convert_shell_script(ctx, script_lines)
-    print("script text: " + script_text)
 
     execution_requirements = {"block-network": ""}
 
     rule_outputs = outputs.declared_outputs + [installdir_copy.file]
 
+    build_script_file = ctx.actions.declare_file("{}_Cc{}MakeRuleScript.sh".format(lib_name, attrs.configure_name.capitalize()))
+    ctx.actions.write(
+        output = build_script_file,
+        content = script_text,
+        is_executable = True,
+    )
+    build_log_file = ctx.actions.declare_file("{}_Cc{}MakeRuleLog.txt".format(lib_name, attrs.configure_name.capitalize()))
+    build_command_lines = [
+        "export BUILD_SCRIPT=\"{}\"".format(build_script_file.path),
+        "export BUILD_LOG=\"{}\"".format(build_log_file.path),
+        "$$BUILD_SCRIPT$$ &> $$BUILD_LOG$$",
```

Comment:
I like the approach, but unfortunately there are couple of problems:

- this file is only for Bazel 0.21, for Bazel 0.22 and newer /new_11_2018/framework.bzl is used
- the current script contains way too much of unix scripting. I would improve the cleanup() function in the shell toolchain to be more abstract so we can use it here, and define a new function for output redirection.

I can do both things myself later, right after applying your PR, so that you do not have to go into so many details.
I am already very grateful for the initiative:)
WDYT?

Reply:
That sounds good to me. Thanks!